src/libCom/osi/os/WIN32/osdThread.c

```diff
@@ -808,14 +808,21 @@ HANDLE osdThreadGetTimer()
 epicsShareFunc void epicsShareAPI epicsThreadSleep ( double seconds )
 {
     static const unsigned nSec100PerSec = 10000000u;
+    static const unsigned mSecPerSec = 1000u;
     LARGE_INTEGER tmo;
     HANDLE timer;
+    LONGLONG nSec100;
 
     if ( seconds <= 0.0 ) {
         tmo.QuadPart = 0u;
     }
+    else if ( seconds >= INFINITE / mSecPerSec  ) {
+        nSec100 = (LONGLONG)(INFINITE - 1) * (nSec100PerSec / mSecPerSec); /* for compatibility with old approach */
+        tmo.QuadPart = -nSec100;
```

Comment:
Trying to decode these expressions gives me:

`INFINITE / mSecPerSec` -> `0xfffffffe/1000` -> `4294967` seconds as the maximum timeout.

`(LONGLONG)(INFINITE - 1) * (nSec100PerSec / mSecPerSec)` -> `0xfffffffeull * 10000` -> `42949672940000` or `4294967.294 ns`

So this tracks.

`tmo.QuadPart = -nSec100;` as `SetWaitableTimer()`[Negative values indicate relative time](https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-setwaitabletimer).

Reply:
I will add a code comment re: negative values